public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -43,35 +111,52 @@ const reduceNamespaces = (filters: FilterState) => {
 };
 
 // Reduces groups to only groups that have rules matching the filters
-const reduceGroups = (filters: FilterState) => {
+const reduceGroups = (ngFilters: SearchFilterState) => {
   return (groupAcc: CombinedRuleGroup[], group: CombinedRuleGroup) => {
     const rules = group.rules.filter((rule) => {
-      if (filters.ruleType && filters.ruleType !== rule.promRule?.type) {
+      if (ngFilters.ruleType && ngFilters.ruleType !== rule.promRule?.type) {
+        return false;
+      }
+      if (
+        ngFilters.dataSourceName &&
+        isGrafanaRulerRule(rule.rulerRule) &&
+        !isQueryingDataSource(rule.rulerRule, ngFilters)
```

Comment:
What do you think of abstracting this condition into a `const`? it's hard to tell what it checks for at a glance.

Reply:
I'll add this. In general, I'd like to refactor the filtering logic, but it's a pretty complex task, and I don't want to include it in this PR